### PR TITLE
Bugfix/2054 lower-bound discrete truncations off by 1

### PR DESCRIPTION
**Pull request by bob-carpenter**
#### Submisison Checklist

- [x] Run unit tests: `./runTests.py src/test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary

Fixes off-by-one error in lower bound truncations for discrete distributions and updates manual to explain discrete vs. continuous translations.

#### How to Verify

Unit tests checking for all combinations of discrete/continuous distributions and built-in/lpdf/log probability function definitions.  Desired output explained and derived in the doc language section.

#### Side Effects

None.

#### Documentation

Yes, updated manual.

#### Reviewer Suggestions

Anyone.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/docs/stan-reference/language.tex [hunk 1]
```diff
@@ -2907,155 +2907,296 @@ \subsection{User-Transformed Variables}
...
+\subsubsection{Truncation with lower and upper bounds in Stan}
 
 Stan allows probability functions to be truncated.  For example, a
-truncated unit normal distribution restricted to $(-0.5, 2.1)$ is
-encoded as follows.
+truncated unit normal distributions restricted to $[-0.5, 2.1]$
+can be coded with the following sampling statement.
 %
 \begin{stancode}
 y ~ normal(0,1) T[-0.5, 2.1];
```
Comment (syclik) [inline src/docs/stan-reference/language.tex]: add space between 0 and 1.

Comment (syclik) [inline src/docs/stan-reference/language.tex]: This pattern is repeated. I can fix them all.

**Comment by davharris** [comment]
Thanks @bob-carpenter! Really looking forward to being able to fit hurdle models again.

**Comment by bob-carpenter** [comment]
You can just code around it with explicit target += statements for now.

This bug wasn't introduced at a late stage---lower bound discrete
truncations never worked properly and it would affect sampling if
the truncated distribution had Stan model parameters (or functions
thereof) as parameters.  

- Bob

**Pull request merged**